controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -198,23 +197,25 @@ def create_volume_returns_error(self, a_enter, create_volume, array_type, return
 
         self.assertEqual(context.code, return_code)
         self.assertTrue(msg in context.details)
-        self.mediator.get_volume.assert_called_once_with(vol_name)
+        self.mediator.get_volume.assert_called_once_with(vol_name, volume_context={'pool': 'pool1'})
         self.mediator.create_volume.assert_called_once_with(vol_name, self.capacity_bytes, {}, self.pool)
 
-    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.detect_array_type")
-    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__enter__")
-    @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.__exit__")
-    def test_create_volume_cuts_name_if_its_too_long(self, a_exit, a_enter, array_type):
-        a_enter.return_value = self.mediator
-        context = utils.FakeContext()
-
-        self.request.name = "a" * 128
-        self.mediator.create_volume = Mock()
-        self.mediator.create_volume.return_value = utils.get_mock_mediator_response_volume(10, "vol", "wwn", "xiv")
-        array_type.return_value = "a9k"
-        res = self.servicer.CreateVolume(self.request, context)
-        self.assertEqual(context.code, grpc.StatusCode.OK)
-        self.mediator.get_volume.assert_called_once_with("a" * self.mediator.max_vol_name_length)
+    # @patch("controller.array_action.array_connection_manager.ArrayConnectionManager.detect_array_type")

```

Comment:
But as specified before you removed the logic also for svc and a9k. Please return kit and return the test.

Reply:
No, I explained the reason of removing it.
volume name comes from pvc name, if the volume name is longer than 63, it means a very long prefix is used. In this case we should raise an error to warn the user instead of cutting the name silently. It is not safe.